### PR TITLE
Feature: ASB-2335 - Re-open register journey based on Re-Open dates

### PR DESCRIPTION
**Pull request by christopher-semmens**
## Describe your changes

https://pins-ds.atlassian.net/browse/ASB-2335?atlOrigin=eyJpIjoiOTRmMWZiZGMwNGVhNDc5MzkxZmM3YmE4NGU5YmEwNGEiLCJwIjoiaiJ9

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/forms-web-app/src/pages/projects/register/index/_utils/get-page-data.js [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
-const {
-	featureFlag: { openRegistrationCaseReferences }
-} = require('../../../../../config');
 
-const getPageData = ({ DateOfRelevantRepresentationClose }, periodOpen, caseRef) => ({
+const getCloseDate = (
+	registrationOpen,
+	registrationOpenCloseDate,
+	registrationReOpened,
+	registrationReOpenedOpenedCloseDate
```
Comment (emil-placheta) [inline packages/forms-web-app/src/pages/projects/register/index/_utils/get-page-data.js:9]: registrationReOpenedOpenedCloseDate? I thought I'm having a stroke

File: packages/forms-web-app/src/pages/projects/register/index/_utils/is-registration-open.js [hunk 2]
```diff
@@ -6,16 +6,50 @@ const { featureFlag } = require('../../../../../config');
...
-	const dayToday = dayjs();
-	const registrationOpenDate = dayjs(openDate);
-	const registrationClosedDate = dayjs(closedDate);
-
-	return (
-		featureFlag.openRegistrationCaseReferences.includes(caseRef) ||
-		(dayToday.isSameOrAfter(registrationOpenDate) &&
-			(dayToday.isSameOrBefore(registrationClosedDate, 'day') || !closedDate))
-	);
+const isRegistrationClosed = ({
```
Comment (emil-placheta) [inline packages/forms-web-app/src/pages/projects/register/index/_utils/is-registration-open.js:9]: small thing, but can we have a comment here explaining that we have `isRegistrationClosed` functions specifically to check for a state where registration is closed after being opened previously and why we need that? It's a bit confusing having two separate functions that check for registration being open and closed which at first glance check the same thing

**Review by emil-placheta — COMMENTED** [review]
2 small things, otherwise LGTM!

**Review by Yakitrak — APPROVED** [review]
LGTM - just a minor comment to add some explanation as @emil-placheta said

File: packages/forms-web-app/src/pages/projects/register/index/_utils/is-registration-open.js (hunk 2 above)
Comment (christopher-semmens) [reply packages/forms-web-app/src/pages/projects/register/index/_utils/is-registration-open.js:15]: @emil-placheta Comment added and argument renamed.

**Pull request merged**